cmake/nlohmann-json.cmake

```diff
@@ -0,0 +1,14 @@
+include(ExternalProject)
+
+ExternalProject_Add(nlohmann-json
```

Comment:
Should just git import the file, shouldn't we?

Reply:
**Please** don't pull in crazy package managers in the build system, things are bad enough as they are :-).
We should just use submodules for this stuff and be done with it. I know there have been complications with using submodules in the past, but IIUC that was due to developing both ends in parallel, which is much more of a hassle.
Pulling in fixed versions of external code that are occasionally updated with submodules is zero hassle.

Reply:
IMHO the only viable alternative to git subtrees or git submodules (of which I'd prefer the latter for this), is just don't do anything at all, document the dependency, have users install it and cmake find it. Everything else is enormously bad and that does *especially* for "dependency management systems". That's the worst you can do to anyone who ever wants to package your package and *that's* who one should be concerned about primarily - regular users rarely build anyways.

Reply:
Just to rant a bit more about this :-D: https://github.com/ethereum/solc-bin/pull/101 is exactly what you get from dependency managers  - it's **so** much more convenient for everyone to just use a random dependency manager (which of course doesn't have like an API and interface of its own you first need to know about, if you ever want to override anything), that it's *totally* worth it to then require people to track down opaque dependency chains with 10 indirections without being able to influence any of it. That's *really* superior to the *extreme* hassle of an occasional ``git submodule update`` :-).

But yeah, sorry :-D. I really think the seeming convenience of dependency managers is an absolute fallacy - it exactly prevents you from hitting problems in cases in which you definitely *should* hit problems as in that you should update your stuff to work with proper up-to-date external dependencies. Even submodules are actually bad in that regard, but for some small header-only dependencies they may be excusable...

Reply:
I think we should continue this discussion in the context of #8860. I quoted the comments there and will post my response in that issue in a moment.